src/zod.ts

```diff
@@ -0,0 +1,211 @@
+		}),
+		delay: z.number().gt(1).lt(1000, {
+			message:
+				"delay is in seconds, >1000 implies you think it's milliseconds",
+		}),
+		action: z.nativeEnum(Action, {
+			invalid_type_error: `action must be either '${Action.SAVE}' or '${Action.INJECT}'`,
+		}),
+		qbittorrentUrl: z.string().url().optional(),
+		rtorrentRpcUrl: z.string().url().optional(),
+		transmissionRpcUrl: z.string().url().optional(),
+
+		matchMode: z.nativeEnum(MatchMode, {
+			invalid_type_error: `matchMode must be either '${MatchMode.RISKY}' or '${MatchMode.SAFE}'`,
+		}),
+		linkType: z.nativeEnum(LinkType, {
+			invalid_type_error: `linkType must have a value of "${LinkType.HARDLINK}" or "${LinkType.SYMLINK}"`,
+		}),
+		dataCategory: z.union([
+			z.undefined(),
+			z.string().min(1).max(24, {
+				message: "dataCategory must have a length of 1-24 characters",
+			}),
+		]),
+		fuzzySizeThreshold: z.number().gt(0).lt(1, {
+			message:
+				"fuzzySizeThreshold must be a decimal percentage greater than 0 and less than 1",
+		}),
+		searchCadence: z.union([
+			z.number(),
+			z.undefined(),
+			z.nan().refine(
+				() => {
+					return typeof fileConfig.searchCadence === "undefined";
+				},
+				{
+					message:
+						"your searchCadence does not follow vercel's `ms` style",
+				}
+			),
+		]),
+		rssCadence: z.union([
+			z.number(),
+			z.undefined(),
+			z.nan().refine(
+				() => {
+					return typeof fileConfig.rssCadence === "undefined";
+				},
+				{
+					message:
+						"your rssCadence does not follow vercel's `ms` style",
+				}
+			),
+		]),
+		excludeOlder: z.union([
+			z.number(),
+			z.undefined(),
+			z.nan().refine(
+				() => {
+					return typeof fileConfig.excludeOlder === "undefined";
+				},
+				{
+					message:
+						"your excludeOlder does not follow vercel's `ms` style",
+				}
+			),
+		]),
+		excludeRecentSearch: z.union([
+			z.number(),
+			z.undefined(),
+			z.nan().refine(
+				() => {
+					return (
+						typeof fileConfig.excludeRecentSearch === "undefined"
+					);
+				},
+				{
+					message:
+						"your excludeRecentSearch does not follow vercel's `ms` style",
+				}
+			),
+		]),
+	})
+	.refine(
+		(config) => {
+			if (config.action === Action.INJECT) {
+				return (
+					config.rtorrentRpcUrl ||
+					config.qbittorrentUrl ||
+					config.transmissionRpcUrl
+				);
+			}
+			return true;
+		},
+		() => ({
+			message:
+				"You need to specify rtorrentRpcUrl, transmissionRpcUrl, or qbittorrentUrl when using 'inject'.",
+		})
+	)
+	.refine(
+		(config) => {
+			if (config.dataDirs || config.linkDir) {
+				return config.dataDirs && config.linkDir && config.linkType;
+			}
+			return true;
+		},
+		() => ({
+			message:
+				"Data-Based Matching requires linkType, dataDirs, and linkDir to be defined.",
+		})
+	)
+	.refine((config) => {
+		if (config.skipRecheck && config.matchMode == MatchMode.RISKY) {
+			logger.warn(
+				"It is strongly recommended to not skip rechecking for risky matching mode"
```

Comment:
I'm not convinced we should be making this recommendation so strongly tbh

Reply:
It is the same recommendation you previously had in validateOptions verbatim, I just put it here instead with the new validation.

What would you want it to say, if anything?